jocd/src/br/org/certi/jocd/dapaccess/Command.java

```diff
@@ -310,8 +365,12 @@ public void add(int count, byte request, long[] words, Byte dapIndex) {
    * Decode the response returned by a DAP_TransferBlock
    * CMSIS-DAP command and return it as an array of bytes.
    */
-  private byte[] decodeTransferBlockData(byte[] data) throws TransferError {
-    assert this.getEmpty() == false;
+  private byte[] decodeTransferBlockData(byte[] data) throws TransferError, Error {
+    // Assert this.getEmpty() == false.
+    if (this.getEmpty()) {
+      throw new Error("decodeTransferBlockData: Unexpected getEmpty() value ( " + getEmpty() + ")");
```

Comment:
This getEmpty() call could be avoided here, since the error state already occurred and it's already known that the return value will be true.

Reply:
Fixed on 3905d3fb99c8c6b556bfe3b84b6ed06c2f55747c